I don't know why but on the compute node I got the error:
```
CMake Error at cmake/SetupPypp.cmake:24 (find_python_module):
  Unknown CMake command "find_python_module".
```
even though `SetupPypp.cmake` where this function is defined is there. When I do it on the entry node it works fine. I don't really understand this, does that tell you anything?

--- a/support/Environments/minerva_gcc.sh
+++ b/support/Environments/minerva_gcc.sh
@@ -3,48 +3,98 @@
 # Distributed under the MIT License.
 # See LICENSE.txt for details.
 
-spectre_setup_modules() {
-    source /home/SPACK/share/spack/setup-env.sh
-}
+# Instructions to **compile** spectre on Minerva:
+#
+# 1. We recommend you compile spectre on a compute node to avoid disrupting the
+#    login node for other users. This is how you can request a compute node on
+#    the `devel` queue for a few hours:
+#    ```sh
+#    srun -p devel --nodes=1 --ntasks-per-node=16 --time=08:00:00 --pty bash -i
+#    ```
+#    You will be dropped right into a shell on the compute node as soon as the
+#    scheduler has allocated one for you so you can proceed with these
+#    instructions.
+# 2. Clone the spectre repository, if you haven't already. A good place is in
+#    your `/home` directory on Minerva:
+#    ```sh
+#    cd /home/yourname
+#    git clone git@github.com:yourname/spectre.git
+#    ```
+# 3. Set the `$SPECTRE_HOME` environment variable to the location of the spectre
+#    repository, e.g. `/home/yourname/spectre`:
+#    ```sh
+#    export SPECTRE_HOME=path/to/spectre
+#    ```
+# 4. Source this script and setup modules:
+#    ```sh
+#    source $SPECTRE_HOME/support/Environments/minerva_gcc.sh`
+#    spectre_setup_modules
+#    ```
+#    Note: Add steps 3 and 4 to your `.bashrc` file if you don't want to repeat
+#    them every time you log in. The `spectre_setup_modules` function only
+#    adjusts your `MODULEPATH` to make the installed modules visible but doesn't
+#    load any, so it's safe to add to your `.bashrc`.
+# 5. Create a build directory, if you don't have one already. A good place is in
+#    your `/work` directory on Minerva:
+#    ```sh
+#    cd /work/yourname
+#    mkdir spectre-build
+#    ```
+#    Note: Add a timestamp or descriptive labels to the name of the build
+#    directory, since you may create more build directories later, e.g.
+#    `build_YYYY-MM-DD` or `build-clang-Debug`.
+# 6. Run `cmake` to configure the build directory:
+#    ```sh
+#    cd path/to/build/directory
+#    module purge
+#    spectre_run_cmake
+#    ```
+#    Note: Remember to `module purge` to work in a clean environment, unless you
+#    have reasons not to.
+# 7. Compile! With the build directory configured, this script sourced and
+#    modules set up, you can skip the previous steps from now on.
+#    ```sh
+#    module purge
+#    spectre_load_modules
+#    make -j16 SPECTRE_EXECUTABLE
+#    ```
+#    Replace `SPECTRE_EXECUTABLE` with the target you want to build, e.g.
+#    `unit-tests` or `SolvePoisson3D`.
+#
+# Instructions to **run** spectre executables on Minerva:
+#
+# 1. Create a run directory. A good place is in your `/scratch` directory on
+#    Minerva. Make sure to choose a descriptive name, e.g.
+#    `/scratch/yourname/spectre/your_project/00_the_run`.
+# 2. Copy `$SPECTRE_HOME/support/SubmitScripts/Minerva.sh` to the run directory
+#    and edit it as the comments in that file instruct.
+# 3. Submit the job to Minerva's queue:
+#    ```sh
+#    sbatch Minerva.sh
+#    ```
 
-spectre_unload_modules() {
-    module unload gcc-7.3.0-gcc-7.3.0-pqzvlyr
-    module unload blaze-3.2-gcc-7.3.0-d4xgiej
-    module unload boost-1.66.0-gcc-7.3.0-rzvaoux
-    module unload brigand-master-gcc-7.3.0-3m5ibui
-    module unload catch-2.2.1-gcc-7.3.0-naocwdf
-    module unload charm-6.8.2-gcc-7.3.0-rqlew72
-    module unload cmake-3.11.4-gcc-7.3.0-3ru42p7
-    module unload gsl-2.4-gcc-7.3.0-yobvdek
-    module unload hdf5-1.10.1-gcc-7.3.0-w32z4ik
-    module unload jemalloc-4.5.0-gcc-7.3.0-wlf2m7r
-    module unload libxsmm-1.9-gcc-7.3.0-npctj4l
-    module unload libsharp-2018-01-17-gcc-7.3.0-4xamgaw
-    module unload openblas-0.3.0-gcc-7.3.0-cjwkbox
-    module unload python-2.7.15-gcc-7.3.0-2azrjn6
-    module unload py-numpy-1.13.3-gcc-7.3.0-ggad2xs
-    module unload py-scipy-1.0.0-gcc-7.3.0-tegl6xx
-    module unload yaml-cpp-develop-gcc-7.3.0-qcfbbll
+spectre_setup_modules() {
+    export MODULEPATH="\
+/home/SPACK2021/share/spack/modules/linux-centos7-haswell:$MODULEPATH"
+    export MODULEPATH="\
+/home/nfischer/spack/share/spack/modules/linux-centos7-haswell:$MODULEPATH"
 }
 
 spectre_load_modules() {
-    module load gcc-7.3.0-gcc-7.3.0-pqzvlyr
-    module load blaze-3.2-gcc-7.3.0-d4xgiej
-    module load boost-1.66.0-gcc-7.3.0-rzvaoux
-    module load brigand-master-gcc-7.3.0-3m5ibui
-    module load catch-2.2.1-gcc-7.3.0-naocwdf
-    module load charm-6.8.2-gcc-7.3.0-rqlew72
-    module load cmake-3.11.4-gcc-7.3.0-3ru42p7
-    module load gsl-2.4-gcc-7.3.0-yobvdek
-    module load hdf5-1.10.1-gcc-7.3.0-w32z4ik
-    module load jemalloc-4.5.0-gcc-7.3.0-wlf2m7r
-    module load libsharp-2018-01-17-gcc-7.3.0-4xamgaw
-    module load libxsmm-1.9-gcc-7.3.0-npctj4l
-    module load openblas-0.3.0-gcc-7.3.0-cjwkbox
-    module load python-2.7.15-gcc-7.3.0-2azrjn6
-    module load py-numpy-1.13.3-gcc-7.3.0-ggad2xs
-    module load py-scipy-1.0.0-gcc-7.3.0-tegl6xx
-    module load yaml-cpp-develop-gcc-7.3.0-qcfbbll
+    module load gcc-10.2.0-gcc-10.2.0-vaerku7
+    module load binutils-2.36.1-gcc-10.2.0-wtzd7wm
+    source /home/nfischer/spack/var/spack/environments/spectre_2021-03-18/loads
+    export CHARM_ROOT="\
+/home/nfischer/spack/opt/spack/linux-centos7-haswell/gcc-10.2.0/\
+charmpp-6.10.2-2waqdh24tz4yt5ozy5clhx6ahxjgivwz"
+    # Libsharp is installed separately with `-fPIC` since the Spack package
+    # doesn't support that option (yet)
+    export LIBSHARP_ROOT="/work/nfischer/spectre/libsharp_2021-03-18/auto"
+}
+
+spectre_unload_modules() {
+    echo "Unloading a subset of modules is not supported."
+    echo "Run 'module purge' to unload all modules."
 }
 
 spectre_run_cmake() {
@@ -53,5 +103,16 @@ spectre_run_cmake() {
         return 1
     fi
     spectre_load_modules
-    cmake -D CMAKE_BUILD_TYPE=Release "$@" $SPECTRE_HOME
+    cmake \
+      -D CMAKE_C_COMPILER=gcc \
+      -D CMAKE_CXX_COMPILER=g++ \
+      -D CMAKE_Fortran_COMPILER=gfortran \
+      -D CHARM_ROOT=$CHARM_ROOT \
+      -D LIBSHARP_ROOT=$LIBSHARP_ROOT \
+      -D CMAKE_BUILD_TYPE=Release \
+      -D DEBUG_SYMBOLS=OFF \
+      -D BUILD_SHARED_LIBS=ON \
+      -D MEMORY_ALLOCATOR=SYSTEM \
+      -D BUILD_PYTHON_BINDINGS=ON \
+      -Wno-dev "$@" $SPECTRE_HOME
 }

--- a/support/SubmitScripts/Minerva.sh
+++ b/support/SubmitScripts/Minerva.sh
@@ -1,0 +1,73 @@
+#!/bin/bash -
+#SBATCH -J spectre
+#SBATCH --nodes 2
+#SBATCH -t 24:00:00
+#SBATCH -p nr
+#SBATCH -o spectre.out
+#SBATCH -e spectre.out
+#SBATCH --ntasks-per-node 16
+#SBATCH --no-requeue
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# To run a job on Minerva:
+#
+# 1. Set the -J, --nodes, and -t options above, which correspond to job name,
+#    number of nodes, and wall time limit in HH:MM:SS, respectively.
+# 2. Set the build directory, run directory, executable name,
+#    and input file below.
+#    Note: The executable will not be copied from the build directory, so if you
+#    update your build directory this file will use the updated executable.
+# 3. Optionally, if you need more control over how SpECTRE is launched on
+#    Minerva you can edit the launch command at the end of this file directly.
+# 4. Submit the job to the queue:
+#    ```sh
+#    sbatch Minerva.sh
+#    ```
+
+# Replace these paths with the path to your build directory and to the
+# directory where you want the output to appear, i.e. the run directory
+export SPECTRE_BUILD_DIR=/work/nfischer/spectre/build_2021-03-18-Release
+export SPECTRE_RUN_DIR=${PWD}
+
+# Choose the executable and input file to run
+# To use an input file in the current directory, set
+# SPECTRE_INPUT_FILE to `${PWD}/InputFileName.yaml` or just `InputFileName.yaml`
+export SPECTRE_EXECUTABLE=SolveXcts
+export SPECTRE_INPUT_FILE=Schwarzschild.yaml
+
+# --- You probably don't need to edit anything below this line ---
+
+mkdir -p ${SPECTRE_RUN_DIR} && cd ${SPECTRE_RUN_DIR}
+
+# Copy the input file into the run directory to preserve it
+cp ${SPECTRE_INPUT_FILE} ${SPECTRE_RUN_DIR}/
+
+# Set up the environment
+export MODULEPATH="\
+/home/SPACK2021/share/spack/modules/linux-centos7-haswell:$MODULEPATH"
+export MODULEPATH="\
+/home/nfischer/spack/share/spack/modules/linux-centos7-haswell:$MODULEPATH"
+module purge
+module load gcc-10.2.0-gcc-10.2.0-vaerku7
+module load binutils-2.36.1-gcc-10.2.0-wtzd7wm
+source /home/nfischer/spack/var/spack/environments/spectre_2021-03-18/loads
+
+# Set permissions for files created with this script
+umask 0022
+
+# Add SpECTRE executables to the path
+export PATH=${SPECTRE_BUILD_DIR}/bin:$PATH
+
+# Run the executable on the available nodes. Notes:
+# - We place only one proc per node because we're running with SMP.
+# - We run on 15 procs per 16-core-node because Charm++ uses one thread per node
+#   for communication.
+# - IntelMPI, OpenMPI, srun and charmrun have different interfaces for similar
+#   things. IntelMPI uses `-n NUM_NODES` and `-ppn PROCS_PER_NODE`, and OpenMPI
+#   uses `-np NUM_NODES` and `--map-by ppr:PROCS_PER_NODE:node`. We are
+#   currently using IntelMPI on Minerva.
+mpirun -n ${SLURM_JOB_NUM_NODES} -ppn 1 \
+  ${SPECTRE_EXECUTABLE} +ppn 15 +pemap 0-14 +commap 15 \
+    --input-file ${SPECTRE_INPUT_FILE}

